src/test/java/io/tackle/pathfinder/controllers/impl/AssessmentsResourceImplTest.java

```diff
@@ -0,0 +1,136 @@
+                @ResourceArg(name = PostgreSQLDatabaseTestResource.USER, value = "pathfinder"),
+                @ResourceArg(name = PostgreSQLDatabaseTestResource.PASSWORD, value = "pathfinder")
+        }
+)
+@QuarkusTestResource(value = KeycloakTestResource.class,
+        initArgs = {
+                @ResourceArg(name = KeycloakTestResource.IMPORT_REALM_JSON_PATH, value = "keycloak/quarkus-realm.json"),
+                @ResourceArg(name = KeycloakTestResource.REALM_NAME, value = "quarkus")
+        }
+)
+public class AssessmentsResourceImplTest extends SecuredResourceTest {
+	@Inject
+	AssessmentSvc assessmentSvc;
+
+	@BeforeEach
+	@Transactional
+	public void init() {
+		System.out.println("Count : " + Assessment.count());
+		System.out.println("DeleteAll : " + Assessment.deleteAll());
+		System.out.println("Count : " + Assessment.count());
+	}
+
+    @Test
+	public void given_ApplicationWithAssessment_When_Get_Then_ReturnsHeaderDto() {
+		assessmentSvc.createAssessment(20L);
+
+		AssessmentHeaderDto[] assessments = given()
+			.queryParam("applicationId", "20")
+		.when()
+			.get("/assessments")
+		.then()
+		   .statusCode(200)
+				.extract().as(AssessmentHeaderDto[].class);
+
+		assertThat(assessments)
+			.hasSize(1)
+			.usingRecursiveComparison()
+				.ignoringFields("id")
+				.isEqualTo(AssessmentHeaderDto.builder().applicationId(20L).notes("").status(AssessmentStatus.STARTED).build());
+  	}
+
+	@Test
+	public void given_ApplicationWithoutAssessment_When_Get_Then_ReturnEmptyAnd200() {
+		Object[] elements = given()
+			.queryParam("applicationId", "30")
+		.when()
+			.get("/assessments")
+		.then()
+			.statusCode(200)
+			.extract()
+			.as(Object[].class);
+		assertThat(elements).isEmpty();
+  	}
+
+	@Test
+	public void given_ApplicationWithoutAssessment_When_CreateAssessment_Then_Returns201() {
+		given()
+			.contentType(ContentType.JSON)
+			.accept(ContentType.JSON)
+			.body(new ApplicationDto(20L))
+		.when()
+			.post("/assessments")
+		.then()
+			.log().all()
+			.statusCode(201)
+			.body("id", greaterThan(0),
+			      "applicationId", equalTo(20),
+				  "status", equalTo("STARTED"));
+	}
+
+	@Test
+	public void given_ApplicationWithAssessment_When_CreateAssessment_Then_Returns400() {
+		assessmentSvc.createAssessment(20L);
+
+		given()
+		    .contentType(ContentType.JSON)
+			.accept(ContentType.JSON)
+			.body(new ApplicationDto(20L))
+		.when()
+			.post("/assessments")
+		.then()
+			.statusCode(400);
+	}
+
+	@Test
+	public void given_NullApplication_When_GetAssessments_Then_Returns400() {
+		given()
+		    .contentType(ContentType.JSON)
+			.accept(ContentType.JSON)
+		.when()
+			.post("/assessments")
+		.then()
+			.statusCode(400);
+	}
+
+	@Test
+	public void given_NullApplication_When_CreateAssessment_Then_Returns400() {
+		given()
+			.contentType(ContentType.JSON)
+			.accept(ContentType.JSON)
+		.when()
+			.post("/assessments")
+		.then()
+			.log().all()
+			.statusCode(400);
+	}	
```

Comment:
The get test method  isn't doing a get and appears to be doing the same call as the create. I think it either needs to test a get or be removed.

Reply:
:+1:  good catch. Method name should be removed.